sdk/state/ingest.go

```diff
@@ -108,3 +115,37 @@ func (c *Channel) ingestTxToUpdateUnauthorizedCloseAgreement(tx *txnbuild.Transa
 
 	return nil
 }
+
+func (c *Channel) ingestTxToUpdateBalances(tx *txnbuild.Transaction) error {
+	localEscrowAddress := c.localEscrowAccount.Address.Address()
+	remoteEscrowAddress := c.remoteEscrowAccount.Address.Address()
+
+	for _, op := range tx.Operations() {
```

Comment:
I suspect we will struggle to cover all cases with this approach of inferring the new balance from the operations in the transaction. This function basically needs to become a stellar-core instance to analyze the tx and execute it.

Instead of inspecting the transaction and trying to figure out from the transaction what the balance of the account has become would it be possible to parse the result/meta XDR that's available for every tx to simply read out the new balance? The meta XDR should contain a before and after snapshot of any accounts and their trust lines that are affected.

I could be wrong about the meta XDR being a feasible approach, but it's what I'd explore and try out.